src/pages/library/datasetBuilder/CohortEditor.test.ts

```diff
@@ -57,10 +72,51 @@ describe('CohortEditor', () => {
       kind: 'list',
       values: [{ id: 0, name: 'value' }],
     });
-    render(createCriteriaViewComponent(_.noop)(criteria));
+    render(
+      h(CriteriaView, {
+        deleteCriteria: _.noop,
+        updateCriteria: _.noop,
+        criteria,
+        key: criteria.id,
+      })
+    );
 
     expect(screen.getByText(criteria.name, { exact: false })).toBeTruthy();
-    expect(screen.getByText(criteria.value.name)).toBeTruthy();
+    expect(screen.getByText(criteria.valuesSelected[0].name, { exact: false })).toBeTruthy();
+  });
+
+  it('updates when list updated', async () => {
+    // Arrange
+    const user = userEvent.setup();
+    const criteria = criteriaFromOption({
+      id: 0,
+      name: 'list',
+      kind: 'list',
+      values: [{ id: 0, name: 'value' }],
+    });
+    let newCriteria = criteriaFromOption({
+      id: 0,
+      name: 'list',
+      kind: 'list',
+      values: [{ id: 0, name: 'value' }],
+    });
+    render(
+      h(CriteriaView, {
+        deleteCriteria: _.noop,
+        updateCriteria: (updatedCriteria) => {
+          newCriteria = updatedCriteria as ProgramDataListCriteria;
+        },
+        criteria,
+        key: criteria.id,
+      })
+    );
+    // Act
+    const select = screen.getByLabelText('Select one or more list');
+    await user.click(select);
```

Comment:
Is this local `select` needed?
```suggestion
    await user.click(screen.getByLabelText('Select one or more list'));
```